x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/esql_request.ts

```diff
@@ -24,29 +42,95 @@ export const performEsqlRequest = async ({
   esClient,
   requestBody,
   requestQueryParams,
+  ruleExecutionLogger,
+  shouldStopExecution,
+  loggedRequests,
 }: {
-  logger?: Logger;
+  ruleExecutionLogger?: IRuleExecutionLogForExecutors;
   esClient: ElasticsearchClient;
-  requestBody: Record<string, unknown>;
+  requestBody: {
+    query: string;
+    filter: QueryDslQueryContainer;
```

Comment:
`wait_for_completion_timeout` and `keep_alive` should probably be params here too

Reply:
I haven't added them, since they were not used. 
Added them now

Reply:
turned out, these values applicable to body only, so I removed them from this type